### PR TITLE
Application constants configuration task.

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**
Application constants configuration task.

### CODE REVIEW

File: src/actions/appConstantsActions.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const loadConstantsConfigSuccessAction = finalConstantsConfig => ({
```
Comment (pepcus-pratik) [inline src/actions/appConstantsActions.js]:
```suggestion
export const loadAppConstantsSuccessAction = constants => ({
```

File: src/actions/appConstantsActions.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  type: 'LOAD_CONSTANTS_CONFIG_SUCCESS_ACTION',
+  appConstants: finalConstantsConfig,
+});
+
+export const loadConstantsConfigFailedAction = errorMessage => ({
```
Comment (pepcus-pratik) [inline src/actions/appConstantsActions.js]:
```suggestion
export const loadAppConstantsFailureAction = errorMessage => ({
```

File: src/apis/core/bootstrap.js [hunk 3]
```diff
@@ -26,3 +26,15 @@ export const getAppTheme = ({ tenant }) => (
...
     url: `src/config/tenant/${tenant}/theme.json`,
   }));
+
+export const getDefaultConstantsConfig = () => (
+  GET({
+    url: 'src/config/tenant/default/constants/text.json',
+  })
+);
+
+export const getConstantsConfig = ({ tenant }) => (
```
Comment (pepcus-pratik) [inline src/apis/core/bootstrap.js]:
```suggestion
export const getAppConstants = ({ tenant }) => (
```

File: src/apis/core/bootstrap.js [hunk 4]
```diff
@@ -26,3 +26,15 @@ export const getAppTheme = ({ tenant }) => (
   GET({
     url: `src/config/tenant/${tenant}/theme.json`,
   }));
+
+export const getDefaultConstantsConfig = () => (
```
Comment (pepcus-pratik) [inline src/apis/core/bootstrap.js]: Don't have a separate function for default constants. Use the same one which is being used for other tenants and just pass the tenant name as **default**

File: src/config/tenant/YJSG-Bhopal/constants/text.json [hunk 5]
```diff
@@ -1 +1,172 @@
...
+
+  "INVALID_ID_MESSAGE": "आपके द्वारा दर्ज किया गया आई.डी. नं. अथवा सीक्रेट कोड गलत है । कृपया पुनः प्रयास करे ।",
+
+  "NO_INFO_CHANGE_MESSAGE": "जानकारी में कोई बदलाव नहीं ",
+
+  "INFO_UPDATE_SUCCESS_MESSAGE": "आपकी जानकारी सफलतापूवर्क अपडेट कर दी गई है |",
+
+  "ADMIN_LOGIN": "Admin Login",
+
+  "USER_TYPES": {
```
Comment (pepcus-pratik) [inline src/config/tenant/YJSG-Bhopal/constants/text.json]: Move this to constants/member.js

File: src/config/tenant/YJSG-Bhopal/constants/text.json [hunk 6]
```diff
@@ -1 +1,172 @@
...
+
+    "MEMBER": "member",
+
+    "ADMIN": "admin",
+
+    "MEMBER_WITH_URL": "member_with_url"
+  },
+  "FILE_PRESENTATION_TYPE": {
```
Comment (pepcus-pratik) [inline src/config/tenant/YJSG-Bhopal/constants/text.json]: Move this to constants/file.js

File: src/config/tenant/YJSG-Bhopal/constants/text.json [hunk 7]
```diff
@@ -1 +1,172 @@
...
+
+    "NORMAL_TABULAR": "normalTabular",
+
+    "COMPLEX_GRID": "complexGrid"
+  },
+  "SUPPORTED_FILE_TYPES": {
```
Comment (pepcus-pratik) [inline src/config/tenant/YJSG-Bhopal/constants/text.json]: Move this to constants/file.js

File: src/config/tenant/YJSG-Bhopal/constants/text.json [hunk 8]
```diff
@@ -1 +1,172 @@
...
+
+    "CSV": "csv",
+
+    "XLSX": "xlsx",
+
+    "XLS": "xls"
+  },
+  "ERROR_MESSAGE_OF_LOAD_APP_FORM_CONFIG": "Error occurred to load form config.",
+
+  "FILES_NAME": {
```
Comment (pepcus-pratik) [inline src/config/tenant/YJSG-Bhopal/constants/text.json]: Move this to constants/file.js

File: src/reducers/constantsConfigReducer.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+const initialState = {
```
Comment (pepcus-pratik) [inline src/reducers/constantsConfigReducer.js]: rename this file to **constants.js**

File: src/reducers/constantsConfigReducer.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  appConstants: {},
+  hasConstantsConfig: false,
+  errorMessage: '',
+};
+
+export const constantsConfigReducer = (state = initialState, action) => {
```
Comment (pepcus-pratik) [inline src/reducers/constantsConfigReducer.js]:
```suggestion
export const constants = (state = initialState, action) => {
```

File: src/reducers/constantsConfigReducer.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  switch (action.type) {
+    case 'LOAD_CONSTANTS_CONFIG_SUCCESS_ACTION': {
+      return {
+        ...state,
+        appConstants: action.appConstants,
+        hasConstantsConfig: true,
```
Comment (pepcus-pratik) [inline src/reducers/constantsConfigReducer.js]: No need to have this flag

File: src/reducers/rootReducer.js [hunk 12]
```diff
@@ -16,6 +17,7 @@ const rootReducer = combineReducers({
   theme,
   appReducer,
   config,
+  constantsConfigReducer,
```
Comment (pepcus-pratik) [inline src/reducers/rootReducer.js]:
```suggestion
  constants,
```

File: src/sagas/core/bootstrap.js [hunk 13]
```diff
@@ -30,6 +37,20 @@ function* getAppConfigurableDataSaga() {
 
 function* getApplicationConstants() {
   const tenant = yield select(getTenantName);
+  const errorMessage = 'Unable to fetch constants config.';
+  try {
```
Comment (pepcus-pratik) [inline src/sagas/core/bootstrap.js]: No need of try-catch

File: src/sagas/core/bootstrap.js [hunk 14]
```diff
@@ -30,6 +37,20 @@ function* getAppConfigurableDataSaga() {
...
+    const defaultConstantsConfig = yield getDefaultConstantsConfig();
+    const tenantConstantsConfig = yield getConstantsConfig({ tenant });
+    const finalConstantsConfig = mergeObjects(defaultConstantsConfig, tenantConstantsConfig);
+    if (finalConstantsConfig) {
```
Comment (pepcus-pratik) [inline src/sagas/core/bootstrap.js]: No need of this condition

File: src/actions/appConstantsActions.js (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/actions/appConstantsActions.js]: Done.

File: src/actions/appConstantsActions.js (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/actions/appConstantsActions.js]: Done.

File: src/apis/core/bootstrap.js (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/apis/core/bootstrap.js]: Done.

File: src/apis/core/bootstrap.js (hunk 4 above)
Comment (BhagyashreeBhoyar) [reply src/apis/core/bootstrap.js]: Done.

File: src/config/tenant/YJSG-Bhopal/constants/text.json (hunk 5 above)
Comment (BhagyashreeBhoyar) [reply src/config/tenant/YJSG-Bhopal/constants/text.json]: Done.

File: src/config/tenant/YJSG-Bhopal/constants/text.json (hunk 6 above)
Comment (BhagyashreeBhoyar) [reply src/config/tenant/YJSG-Bhopal/constants/text.json]: Done.

File: src/config/tenant/YJSG-Bhopal/constants/text.json (hunk 7 above)
Comment (BhagyashreeBhoyar) [reply src/config/tenant/YJSG-Bhopal/constants/text.json]: Done.

File: src/config/tenant/YJSG-Bhopal/constants/text.json (hunk 8 above)
Comment (BhagyashreeBhoyar) [reply src/config/tenant/YJSG-Bhopal/constants/text.json]: Done.

File: src/reducers/constantsConfigReducer.js (hunk 9 above)
Comment (BhagyashreeBhoyar) [reply src/reducers/constantsConfigReducer.js]: Done.

File: src/reducers/constantsConfigReducer.js (hunk 10 above)
Comment (BhagyashreeBhoyar) [reply src/reducers/constantsConfigReducer.js]: Done.

File: src/reducers/constantsConfigReducer.js (hunk 11 above)
Comment (BhagyashreeBhoyar) [reply src/reducers/constantsConfigReducer.js]: Done.

File: src/reducers/rootReducer.js (hunk 12 above)
Comment (BhagyashreeBhoyar) [reply src/reducers/rootReducer.js]: Done.

File: src/sagas/core/bootstrap.js (hunk 13 above)
Comment (BhagyashreeBhoyar) [reply src/sagas/core/bootstrap.js]: Done.

File: src/sagas/core/bootstrap.js (hunk 14 above)
Comment (BhagyashreeBhoyar) [reply src/sagas/core/bootstrap.js]: Done.

**Pull request merged**